SpatialGDK/Source/SpatialGDK/Private/Utils/InterestFactory.cpp

```diff
@@ -219,6 +221,47 @@ void InterestFactory::AddServerSelfInterest(Interest& OutInterest, const Worker_
 	AddComponentQueryPairToInterestComponent(OutInterest, SpatialConstants::ENTITY_ACL_COMPONENT_ID, LoadBalanceQuery);
 }
 
+bool InterestFactory::CheckOwnersHaveEntityId(const AActor* Actor) const
+{
+	AActor* Owner = Actor->GetOwner();
+
+	while (Owner != nullptr && !Owner->IsPendingKillPending())
+	{
+		if (PackageMap->GetEntityIdFromObject(Owner) == SpatialConstants::INVALID_ENTITY_ID)
+		{
+			return false;
+		}
+		Owner = Owner->GetOwner();
+	}
+
+	return true;
+}
+
+void InterestFactory::AddOwnerInterestOnServer(Interest& OutInterest, const AActor* InActor) const
+{
+	AActor* Owner = InActor->GetOwner();
+	Query OwnerChainQuery;
+	
+	while (Owner != nullptr && !Owner->IsPendingKillPending())
+	{
+		QueryConstraint OwnerQuery;
+		OwnerQuery.EntityIdConstraint = PackageMap->GetEntityIdFromObject(Owner);
+		if (OwnerQuery.EntityIdConstraint == SpatialConstants::INVALID_ENTITY_ID)
+		{
+			return ;
```

Comment:
Can we add a log that this means the interest is currently invalid, and will be handled by the `SetNeedOwnerInterestUpdate` called in `SpatialActorChannel`